### PR TITLE
Infere install_type and various cleanups found during integration testing

### PR DESCRIPTION
**Pull request by evanphx**
Using these a bit it's too easy to set everything and forget to change install_type. Originally I was going to make install_type a required variable but then realized that just infering it was probably better.

This also fixes a bug where kubeadm fails if the prefix has any upcase characters.

### CODE REVIEW

File: config.tf [hunk 1]
```diff
@@ -1,6 +1,12 @@
+locals {
+  demo_or_es   = "${var.postgresql_address != "" ? "external_services" : "demo"}"
+  ag_or_ol     = "${var.airgap_package_url != "" ? "airgap" : "online" }"
```
Comment (erindatkinson) [inline config.tf]: if we change this to something like 
```suggestion
  app_network_type     = "${var.airgap_package_url != "" ? "airgap" : "online" }"
```

or something where we say what we're using it for, the variable would be easier to understand.

File: config.tf [hunk 2]
```diff
@@ -1,6 +1,12 @@
+locals {
+  demo_or_es   = "${var.postgresql_address != "" ? "external_services" : "demo"}"
```
Comment (erindatkinson) [inline config.tf]: same comment as https://github.com/hashicorp/terraform-aws-terraform-enterprise/pull/18/files#r337215666

but for something like 
```suggestion
  app_data_mode   = "${var.postgresql_address != "" ? "external_services" : "demo"}"
  
  or something to say what it is better.
```

File: replicated.tf [hunk 3]
```diff
@@ -1,15 +1,15 @@
 locals {
   rptfeconf = {
-    "demo"      = "${file("${path.module}/data/demo.json")}"
-    "airgap"    = "${file("${path.module}/data/airgap.json")}"
-    "es"        = "${file("${path.module}/data/es.json")}"
-    "es_airgap" = "${file("${path.module}/data/es_airgap.json")}"
+    "demo-online"              = "${file("${path.module}/data/demo.json")}"
```
Comment (erindatkinson) [inline replicated.tf]: at some point(not GA) I think we could leverage terraform's templating language better with these

File: config.tf [hunk 4]
```diff
@@ -1,6 +1,14 @@
+locals {
+  app_data_mode   = "${var.postgresql_address != "" ? "external_services" : "demo"}"
+  
+  or something to say what it is better.
```
Comment (erindatkinson) [inline config.tf]: sorry, the suggestion got messed up

File: config.tf [hunk 5]
```diff
@@ -1,6 +1,14 @@
...
+  app_network_type     = "${var.airgap_package_url != "" ? "airgap" : "online" }"
+  install_type = "${local.demo_or_es}-${local.ag_or_ol}"
```
Comment (erindatkinson) [inline config.tf]: this also needs to be changed if you update the previous two

Comment (evanphx) [reply config.tf]: Yup, done.

File: replicated.tf (hunk 3 above)
Comment (evanphx) [reply replicated.tf]: Probably, yeah.

**Pull request merged**